### ISSUE TITLE
Addon configs (just an idea for now)

### ISSUE THREAD

**Issue by aeskathon** [Excrulon/Tree-of-Savior-Lua-Mods#117]
Editing lua files is bad — you need to rerun the whole game for the changes to take effect.
Thus it would be nice to have an ingame configurer at least for current session (I am not sure if anything can be stored between game launches - is access to file system allowed?).
I believe something has to be done before there is a whole zoo of different addons.
Ideally it should be some kind of UI, but for now command line configuration would suffice (with keeping the the idea of UI config in mind).
So every addon must expose some kind of API allowing to configure it from another addon, so you can put the initial configuration in one place (for example addonloader.lua) and create another addon for ingame configuration.
I guess the API should be like:
GetConfigurableVariables() — returns list of variable names
GetConfigValue("variable_name")
SetConfigValue("variable_name", value)
For UI reasons we could also have
GetConfigDescription("variable_name") and GetConfigPossibleValues("variable_name") though I am not sure what the latter should return.

What do you guys think?

**Comment by Excrulon** [comment]
yeah yeah it's all just time.

**Comment by wickedDesigns** [comment]
Seems fairly reasonable.  I'd imagine this could be accessed via another icon in the system menu.

**Comment by Excrulon** [comment]
It's up to the addon author to implement how they want their settings set. Will be using messagepack for mine as I go.
